cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -893,7 +893,8 @@ class TableSorter {
 
   TableSorter(ExecContext* ctx, uint64_t* indices_begin, uint64_t* indices_end,
               const Table& table, const SortOptions& options)
-      : ctx_(ctx),
```

Comment:
Yes, it was segfaulting when it was resetting the error because it was trying to delete an uninitialized object (or something like that, I fixed it a while ago and don't remember the exact details).

Reply:
not in any very detailed way, just that it was inside this constructor, and there was some shared_ptr being deleted. it also didn't always reproduce, but it stopped reproducing once i made status be initialized before it got passed around to other stuff.